src/ctx.rs

```diff
@@ -23,54 +78,85 @@ struct PersistedVariable {
+        variables
             .tree()
             .iter()
             .map(|(k, v)| PersistedVariable {
                 name: k.into(),
                 default_value: v.clone(),
             })
-            .collect::<Vec<PersistedVariable>>();
-        PersistedOptions {
-            variables: formatted_variables,
-        }
+            .collect::<Vec<PersistedVariable>>()
     }
 }
 
-pub fn extract_variables(ctx: &Ctx) -> Result<(Variables, Variables)> {
-    let mut confirmed_variables = Variables::default();
-    confirmed_variables.insert(
+pub(crate) fn extract_variables(ctx: &Ctx) -> Result<(Variables, Variables, Variables)> {
+    let mut default_variables = Variables::default();
+    default_variables.insert(
         "ffizer_dst_folder",
         ctx.cmd_opt
             .dst_folder
             .to_str()
             .expect("dst_folder to converted via to_str"),
     )?;
-    confirmed_variables.insert("ffizer_src_uri", ctx.cmd_opt.src.uri.raw.clone())?;
-    confirmed_variables.insert("ffizer_src_rev", ctx.cmd_opt.src.rev.clone())?;
-    confirmed_variables.insert("ffizer_src_subfolder", ctx.cmd_opt.src.subfolder.clone())?;
-    confirmed_variables.insert("ffizer_version", env!("CARGO_PKG_VERSION"))?;
+    default_variables.insert("ffizer_src_uri", ctx.cmd_opt.src.uri.raw.clone())?;
+    default_variables.insert("ffizer_src_rev", ctx.cmd_opt.src.rev.clone())?;
+    default_variables.insert("ffizer_src_subfolder", ctx.cmd_opt.src.subfolder.clone())?;
+    default_variables.insert("ffizer_version", env!("CARGO_PKG_VERSION"))?;
 
-    confirmed_variables.append(&mut get_cli_variables(ctx)?);
+    let confirmed_variables = get_cli_variables(ctx)?;
     let suggested_variables = get_saved_variables(ctx)?;
 
-    Ok((confirmed_variables, suggested_variables))
+    Ok((default_variables, confirmed_variables, suggested_variables))
 }
 
-pub fn save_options(variables: &Variables, ctx: &Ctx) -> Result<()> {
+fn key_from_loc(source: &SourceLoc) -> (String, String, String) {
+    let uri = &source.uri;
+    (
+        uri.host.clone().unwrap_or_default(),
+        uri.path.to_string_lossy().into(),
+        source
+            .subfolder
+            .clone()
+            .unwrap_or_default()
+            .to_string_lossy()
+            .into(),
+    )
+}
+
+pub(crate) fn save_options(variables: &Variables, ctx: &Ctx) -> Result<()> {
+    // Save or update default variable values stored in datastore
+    let mut variables_to_save = get_saved_variables(ctx)?;
+    variables_to_save.append(&mut variables.clone()); // update already existing keys
+    variables_to_save.retain(|k, _v| !k.starts_with("ffizer_"));
+
+    let new_key = key_from_loc(&ctx.cmd_opt.src);
+
+    let mut saved_srcs: Vec<PersistedSrc> = get_saved_sources(&ctx.cmd_opt.dst_folder)?
+        .into_iter()
+        .filter(|loc| key_from_loc(loc) != new_key)
+        .map(|loc| loc.into())
+        .collect();
+
+    saved_srcs.push(ctx.cmd_opt.src.clone().into());
```

Comment:
If you want to follow the same order than the `import` section in `.ffizer.yaml`, then the latest applied should be the first in the list.